src/graphql/handler.rs

```diff
@@ -1,34 +1,37 @@
-use crate::database::pool::PgPool;
-use crate::graphql::manager::create_context;
-use crate::graphql::model::Schema;
-use actix_web::{web, Error, HttpResponse};
-use futures::Future;
-use juniper::http::GraphQLRequest;
-use std::sync::Arc;
-extern crate serde_json;
+use crate::cli_args::Opt;
+use crate::database::{db_connection, Pool};
+use crate::graphql::model::{Context, Schema};
 use crate::jwt::model::DecodedToken;
-use crate::serde::ser::Error as SerdeError;
 use crate::user::model::LoggedUser;
+use actix_web::{error, web, Error, HttpResponse};
+use juniper::http::graphiql::graphiql_source;
+use juniper::http::GraphQLRequest;
+use std::sync::Arc;
 
-pub fn graphql_handler(
+pub(super) async fn graphql(
     st: web::Data<Arc<Schema>>,
     data: web::Json<GraphQLRequest>,
     user: LoggedUser,
     token: DecodedToken,
-    pool: web::Data<PgPool>,
-) -> impl Future<Item = HttpResponse, Error = Error> {
-    web::block(move || {
-        let pg_pool = pool.get().map_err(|e| serde_json::Error::custom(e))?;
+    pool: web::Data<Pool>,
+    opt: web::Data<Opt>,
+) -> Result<HttpResponse, Error> {
+    let db_pool = db_connection(&pool)?;
```

Comment:
The factorization of `db_connection` allows us simply use the `?` operator here, and keep the following lines easier to read / maintain instead of chains with closures.